src/main/kotlin/org/jetbrains/research/libsl/visitors/ExpressionVisitor.kt

```diff
@@ -486,10 +492,27 @@ class ExpressionVisitor(
     }
 
     override fun visitCallAutomatonConstructorWithNamedArgs(
-        ctx: LibSLParser.CallAutomatonConstructorWithNamedArgsContext
+        ctx: CallAutomatonConstructorWithNamedArgsContext
     ): Expression {
         val automatonName = ctx.name.asPeriodSeparatedString()
         val automatonRef = AutomatonReferenceBuilder.build(automatonName, context)
+
+        val generics = processGenerics(ctx.generic())
+
+        generics.forEach {
+            if (!GenericTypeBound.EMPTY.equals(it.typeBound) || "?".equals(it.name))
+                // TODO: add for all exceptions in parser concrete places where it was appeared.
+                throw error("Constructor invoke can't contain WildCards")
+
+        }
+        
+        typeOfVariable?.genericReferences?.forEach {
+            if (!GenericTypeBound.EMPTY.equals(it.typeBound) || "?".equals(it.name))
+            // TODO: add for all exceptions in parser concrete places where it was appeared.
+                throw error("Type of variable which assigns to invokation of the constructor can't contain generics with wildcard int type")
+
+        }
```

Comment:
I am a bit confused. Why do we need to check the variable type during automaton construction?
I think these use-cases should be disallowed:
```dart
var foo: Collection<?> = new AutoCollection<?>(state=...);
var bar: Collection<?> = new AutoCollection<in String>(state=...);
```
but these -- allowed:
```dart
var fizz: Collection<?> = new AutoCollection<Int>(state=...);
var buzz: Array<out Number> = new AutoArray<Byte>(state=...);
```
*(note: attention to the automaton construction expression, not the variable declaration part)*

Reply:
Fixed;